### PR TITLE
[Bug][FieldCollection]: Fixes potential relation data loss

### LINKED ISSUE #15448: [Bug]: [FieldCollection] Data Loss if Relation has been deleted
### Pimcore version

10.5.x

### Steps to reproduce

If you're removing relation(s) from a Fieldcollection via PHP API, all relations will be removed after resaving the object.

Steps to reproduce:
```php

$c = MyObject::getById(100);

$fc = new Fieldcollection();

$asset1 = new Asset();
$asset1->setParent(Asset::getByPath('/'));
$asset1->setKey('test-1');
$asset1->save();

$asset2 = new Asset();
$asset2->setParent(Asset::getByPath('/test'));
$asset2->setKey('test-2');
$asset2->save();

$fcb1 = new Fieldcollection\Data\DocumentsBlock();
$fcb1->setIdentifier('test1');
$fcb1->setDocuments([new ElementMetadata('documents', [], $asset1)]);

$fcb2 = new Fieldcollection\Data\DocumentsBlock();
$fcb2->setIdentifier('test2');
$fcb2->setDocuments([new ElementMetadata('documents', [], $asset2)]);

$fc->add($fcb1);
$fc->add($fcb2);

$c->setDocuments($fc);
$c->save(); // result: see Screenshot I

// now delete a relation
$asset2->delete();

$freshC = MyObject::getById(100, ['force' => true]);
$freshC->save();  // result: see Screenshot II
```

## Screenshot I
image

## Screenshot II
image

BTW: Didn't check it on 11, but i guess it's the same there.

### Expected / actual behavior

Only the relation gets removed, other fields should not be touched.

### PR DESCRIPTION
**Pull request by kingjia90**
## Changes in this pull request  
Resolves https://github.com/pimcore/pimcore/issues/15448

## Additional info
It looks like it's missing this 
https://github.com/pimcore/pimcore/blob/9835d182b6456edb1b2a0ea569d9a98701264523/models/DataObject/ClassDefinition/Data/Relations/ManyToManyRelationTrait.php#L26-L28

See also https://github.com/pimcore/pimcore/issues/6733

### <samp>🤖 Generated by Copilot at dfec489</samp>

Added an option to force saving advanced many-to-many relations that are changed outside the object. Improved code style in `AdvancedManyToManyRelation.php`.

### <samp>🤖 Generated by Copilot at dfec489</samp>

> _`forceSave` option_
> _lets relation data change_
> _autumn leaves falling_

### <samp>🤖 Generated by Copilot at dfec489</samp>

*  Add `forceSave` parameter to `save` function of `AdvancedManyToManyRelation` class to allow bypassing dirty detection logic ([link](https://github.com/pimcore/pimcore/pull/15483/files?diff=unified&w=0#diff-1d0bb2ae86274681b1012717ccb5958660cfdaf177d0c2a3e5f6b1cf039dbf98L552-R564))
*  Format code by adding blank line in `models/DataObject/ClassDefinition/Data/AdvancedManyToManyRelation.php` ([link](https://github.com/pimcore/pimcore/pull/15483/files?diff=unified&w=0#diff-1d0bb2ae86274681b1012717ccb5958660cfdaf177d0c2a3e5f6b1cf039dbf98R570))

### CODE REVIEW

**Review by dvesh3 — COMMENTED** [review]
Looks good from the initial code scan. we should definately adapt [tests](https://github.com/pimcore/pimcore/blob/44dd0bf7b9a016b292ce6cf7d5da2894d6eac4da/tests/Model/Relations/FieldcollectionTest.php#L33) if it doesn't cover the scenarios reported.

**Review by dvesh3 — COMMENTED** [review]
At the moment, these checks are scattered at different places. see below:
https://github.com/pimcore/pimcore/blob/2f18a0f26380c56b609d937b40d2050c8d1a3e1a/models/DataObject/ClassDefinition/Data/Relations/ManyToManyRelationTrait.php#L28
https://github.com/pimcore/pimcore/blob/75bbc5bbacbd57383ccb20e52bbb151397b81a78/models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php#L98
https://github.com/pimcore/pimcore/blob/5cc5f8268182262307476cce0f2646739d44beff/models/DataObject/ClassDefinition/Data/AdvancedManyToManyRelation.php#L552
https://github.com/pimcore/pimcore/blob/7888ed7a38551f5b74671f97457ef863f728463f/models/DataObject/ClassDefinition/Data/AdvancedManyToManyObjectRelation.php#L476

which makes it a bit harder to properly maintain. We should refactor it and make it re-usable.

File: models/DataObject/ClassDefinition/Data/Relations/ManyToManyRelationTrait.php [hunk 1]
```diff
@@ -21,29 +21,43 @@
...
     /**
-     * {@inheritdoc}
+     * Unless forceSave is set to true, this method will check if the field is dirty and skip the save if not
+     *
+     * @param object $object
+     * @param array $params
+     * @return bool
      */
-    public function save($container, $params = [])
+    protected function skipSaveCheck(object $object, $params = []): bool
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/Relations/ManyToManyRelationTrait.php]:
```suggestion
    protected function skipSaveCheck(object $object, array $params = []): bool
```

File: tests/Model/Relations/FieldcollectionTest.php [hunk 2]
```diff
@@ -118,6 +120,57 @@ public function testRelationFieldInsideFieldCollection()
         $this->assertEquals([], $rel);
     }
 
+    public function testAdvancedRelationFieldInsideFieldCollection()
```
Comment (blankse) [inline tests/Model/Relations/FieldcollectionTest.php]:
```suggestion
    public function testAdvancedRelationFieldInsideFieldCollection(): void
```

**Review by dvesh3 — APPROVED** [review]

**Pull request merged**